controllers/cluster_controller.go

```diff
@@ -1097,6 +1097,39 @@ type localityInfo struct {
 
 	// The locality map.
 	LocalityData map[string]string
+
+	Class fdbtypes.ProcessClass
+}
+
+// These indexes are used for sorting and since we sort ascending
+func getClassIndex(cls fdbtypes.ProcessClass) int {
+	switch cls {
```

Comment:
I'm nervous about the possibility for the non-exhaustiveness to introduce non-determinism down the line. When we're using this, maybe we could fall back to a direct string comparison for unknown process classes?

Reply:
So there are two things we need to keep in mind:

1.) Currently the coordinator selection will try storage -> log -> transaction processes all other processes are currently ignore so in theory the default case should never happen.
2.) If the default case will happen this only means that all processes that are not storage/log/transaction will have the least preference (have the highest index) and all processes in that "bucket" will be sorted by their ID so the behaviour is deterministic.

Reply:
You're right, since we're sorting by IP we don't have a risk of non-determinism.